vumi/transports/httprpc/httprpc.py

```diff
@@ -193,6 +195,16 @@ def setup_transport(self):
             ],
             self.web_port)
 
+        self.status_detect = StatusEdgeDetector()
+
+    def add_status(self, **kw):
+        '''Publishes a status if it is not a repeat of the previously
+        published status.'''
+        status = TransportStatus(**kw)
+        if self.status_detect.check_status(status):
+            return self.publish_status(**kw)
+        return succeed(None)
```

Comment:
It's not strictly necessary, I just think it's nice to always return a deferred.